### ISSUE TITLE
Issue when watched key is regenerated

### ISSUE THREAD

**Issue by popsikle** [kelseyhightower/confd#205]
when using watch mode if the keys you are watching get rm'd and replaced conf.d seems to get stuck in a loop watching the old key/index.

Normal logs
```
2015-02-18T19:13:03Z srv-01.o.keep.wtf confd[5535]: DEBUG Retrieving keys from store
2015-02-18T19:13:03Z srv-01.o.keep.wtf confd[5535]: DEBUG Key prefix set to /ci-west
2015-02-18T19:13:03Z srv-01.o.keep.wtf confd[5535]: DEBUG Using source template /etc/confd/templates/crick_api.tmpl
2015-02-18T19:13:03Z srv-01.o.keep.wtf confd[5535]: DEBUG Compiling source template /etc/confd/templates/crick_api.tmpl
2015-02-18T19:13:03Z srv-01.o.keep.wtf confd[5535]: DEBUG Comparing candidate config to /etc/nginx/conf.d/crick_api.conf
2015-02-18T19:13:03Z srv-01.o.keep.wtf confd[5535]: DEBUG Target config /etc/nginx/conf.d/crick_api.conf in sync
2015-02-18T19:13:08Z srv-01.o.keep.wtf confd[5535]: DEBUG Retrieving keys from store
2015-02-18T19:13:08Z srv-01.o.keep.wtf confd[5535]: DEBUG Key prefix set to /ci-west
2015-02-18T19:13:08Z srv-01.o.keep.wtf confd[5535]: DEBUG Using source template /etc/confd/templates/crick_api.tmpl
2015-02-18T19:13:08Z srv-01.o.keep.wtf confd[5535]: DEBUG Compiling source template /etc/confd/templates/crick_api.tmpl
2015-02-18T19:13:08Z srv-01.o.keep.wtf confd[5535]: DEBUG Comparing candidate config to /etc/nginx/conf.d/crick_api.conf
2015-02-18T19:13:08Z srv-01.o.keep.wtf confd[5535]: DEBUG Target config /etc/nginx/conf.d/crick_api.conf in sync
```
Deploy of new app rm's a bunch of etcd keys and regenerates them right here, and confd never recovers
```
2015-02-18T19:23:08Z srv-01.o.keep.wtf confd[5535]: ERROR 401: The event in requested index is outdated and cleared (the requested history has been cleared [5976099/5973919]) [5977098]
2015-02-18T19:23:10Z srv-01.o.keep.wtf confd[5535]: ERROR 401: The event in requested index is outdated and cleared (the requested history has been cleared [5976100/5973919]) [5977099]
```

**Comment by ndarilek** [comment]
Seeing this here as well. Is there a fix in sight? Unfortunately it seems to negate the benefits of confd if things aren't getting synced.

**Comment by popsikle** [comment]
@kelseyhightower I know you are super busy, but you think this is any issue in confd or the etcd library?

**Comment by bacongobbler** [comment]
I think it looks like an issue in etcd. The stack trace is as follows in confd:

https://github.com/kelseyhightower/confd/blob/2ad79d60cd9c284a379b13364322021158efa998/confd.go#L52
https://github.com/kelseyhightower/confd/blob/2ad79d60cd9c284a379b13364322021158efa998/resource/template/processor.go#L96
https://github.com/kelseyhightower/confd/blob/2ad79d60cd9c284a379b13364322021158efa998/backends/etcd/client.go#L77-L79

**Comment by bacongobbler** [comment]
closed via #254
